File: 1055 The World's Richest.cpp
#include<cstdio>
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;

int n, k;

struct Stu{
    string name;
    int age;
    int w;
} s[100001];

bool cmp(Stu a, Stu b){
    if (a.w != b.w) return a.w > b.w;
    else if (a.age != b.age) return a.age < b.age;
    else return a.name < b.name;
}

void query(int m, int amin, int amax){
    int i, cnt=0;
    
    for (i=0; i<n && cnt<m; i++){
        if (s[i].age >= amin && s[i].age <= amax){
            printf("%s %d %d\n", s[i].name.c_str(), s[i].age, s[i].w);
            cnt++;
        }
    }
    
    if (cnt == 0){
        cout << "None" << endl;
    }
}

int main(){
    int i;
    cin>>n>>k;
    
    for (i=0; i<n; i++){
        cin >> s[i].name >> s[i].age >> s[i].w;
    }
    
    sort(s, s+n, cmp);
    
    for (i=1; i<k+1; i++){
        int x, amin, amax;
        
        cin >> x >> amin >> amax;
        
        printf("Case #%d:\n", i);
        query(x, amin, amax);
    }
    
    return 0;
}

File: 1055 The World's Richest.py
N, K = map(int, input().split())
info = []

def query(m, amin, amax):
    l = list(filter(lambda x: amin<=x[1]<=amax, info))

    if not l:
        print("None")
        return
    
    if m > len(l):
        m = len(l)
    for i in range(m):
        print(l[i][0], l[i][1], l[i][2])

if __name__ == "__main__":
    for _ in range(N):
        name, age, w = input().split()
        info.append((name, int(age), int(w)))
    
    info.sort(key=lambda x: (-x[2], x[1], x[0]))

    for i in range(1, K+1):
        print(f"Case #{i}:")
        query(*map(int, input().split()))


'''
两个测试点超时...
'''